### PR TITLE
[WIP] Update cuDF Intro Notebook

### PR DESCRIPTION
**Pull request by beckernick**
Still annoying to comment on notebooks on github 😞

### CODE REVIEW

File: cudf/Intro to cuDF-Copy1.ipynb [hunk 1]
```diff
@@ -0,0 +1,1628 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Intro to cuDF\n",
+    "=======================\n",
+    "\n",
+    "This is a short introduction to cuDF, partly modeled after 10 Minutes to Pandas, t"
```
Comment (beckernick) [inline cudf/Intro to cuDF-Copy1.ipynb]: Missing part of this sentence

File: cudf/Intro to cuDF-Copy1.ipynb [hunk 2]
```diff
@@ -0,0 +1,1628 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a id=\"pandas\"></a>\n",
+    "## Pandas\n",
+    "\n",
+    "Data scientists typically work with two types of data: unstructured and structured. Unstructured data often comes in the form of text, images, or videos. Structured data - as the name suggests - comes in a structured form, often represented by a table or CSV. We'll focus the majority of these tutorials on working with these types of data.\n",
```
Comment (beckernick) [inline cudf/Intro to cuDF-Copy1.ipynb]: "these types" is ambiguous. Let's be explicit about the type

File: cudf/Intro to cuDF-Copy1.ipynb [hunk 3]
```diff
@@ -0,0 +1,1628 @@
...
+   "metadata": {},
+   "source": [
+    "Note how the syntax for both creating and manipulating a cuDF DataFrame is identical to the syntax necessary to create and manipulate Pandas DataFrames; the cuDF API is based on the Pandas API. This design choice minimizes the cognitive burden of switching from a CPU based workflow to a GPU based workflow and allows data scientists to focus on solving problems while benefitting from the speed of a GPU!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# DataFrame Basics with cuDF"
```
Comment (beckernick) [inline cudf/Intro to cuDF-Copy1.ipynb]: Let's add a small intro here as well, to describe the tutorial that attendees are about to experience.

**Comment by beckernick** [comment]
Changed my mind about this process @asmeurer . Just going to use this PR to change the original notebook instead of comment

File: cudf/Intro to cuDF.ipynb [hunk 4]
```diff
@@ -584,46 +619,85 @@
...
+    "Supported logical operators include `>`, `<`, `>=`, `<=`, `==`, and `!=`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Exercise 2\n",
+    "\n",
+    "Try to select only the rows from `df` where the value in column `b` is greater than the vaue in column `c` + 6.\n",
```
Comment (asmeurer) [inline cudf/Intro to cuDF.ipynb]:
```suggestion
    "Try to select only the rows from `df` where the value in column `b` is greater than the value in column `c + 6`.\n",
```

File: cudf/Intro to cuDF.ipynb [hunk 5]
```diff
@@ -712,81 +824,134 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Exercise 3\n",
+    "\n",
+    "Try to convert all the strings to uppercase. Take a look at the nvStrings API documentation linked above if you need some help.\n",
+    "\n",
+    "<details><summary><b>Solution</b></summary>\n",
+    "   <pre>\n",
+    "    <br>print(df.query(\"b > c + 6\"))\n",
```
Comment (asmeurer) [inline cudf/Intro to cuDF.ipynb]: Wrong solution

File: cudf/Intro to cuDF.ipynb [hunk 6]
```diff
@@ -814,122 +979,129 @@
...
-    "df_a['key'] = [0, 1, 2, 3, 4]\n",
-    "df_a['vals_a'] = [float(i + 10) for i in range(5)]\n",
-    "\n",
-    "df_b = cudf.DataFrame()\n",
-    "df_b['key'] = [1, 2, 4]\n",
-    "df_b['vals_b'] = [float(i+10) for i in range(3)]\n",
-    "\n",
-    "df_merged = df_a.merge(df_b, on=['key'], how='left')\n",
-    "print(df_merged.sort_values('key'))"
+    "print(s.append(s))"
```
Comment (asmeurer) [inline cudf/Intro to cuDF.ipynb:1035]: This fails for me. I guess it's because I'm using cudf 0.7?

Comment (beckernick) [reply cudf/Intro to cuDF.ipynb:1035]: Ah, didn't think about that. It's due to the nulls. The fix for this is done, but it won't be "officially released" until 0.8

**Comment by asmeurer** [comment]
Some of the markdown cells mention exact timing outputs. We should probably rerun the notebook on the cloud once that is set up so we can reference the timings in those machines, which should be roughly the same as what the attendees see.

File: cudf/Intro to cuDF.ipynb (hunk 6 above)
Comment (beckernick) [reply cudf/Intro to cuDF.ipynb:1035]: We'll go with 0.8 based on slack discussion

**Pull request merged**